tiledb/api/c_api/fragments_list/fragments_list_api.cc

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file tiledb/api/c_api/fragments_list/fragments_list_api.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines the fragments list section of the C API for TileDB.
+ */
+
+#include "fragments_list_api_external.h"
+#include "fragments_list_api_internal.h"
+#include "tiledb/api/c_api_support/c_api_support.h"
+
+namespace tiledb::api {
+
+capi_return_t tiledb_fragments_list_get_fragment_uri(
+    tiledb_fragments_list_t* f, int index, const char** uri) {
+  ensure_fragments_list_is_valid(f);
+  ensure_output_pointer_is_valid(uri);
+
+  try {
+    *uri = f->get_fragment_uri(index).c_str();
+  } catch (StatusException& e) {
+    return TILEDB_ERR;
+  }
+  return TILEDB_OK;
```

Comment:
Can omit this `try` block entirely. The exception wrapper at the top level will handle it.